lib/iris/experimental/regrid.py

```diff
@@ -1312,3 +1313,401 @@ def regridder(self, src_grid, target_grid):
+                msg = 'Cannot update aux_factory {!r} because of dropped' \
+                      ' coordinates.'.format(factory.name())
+                warnings.warn(msg)
+        return result
+
+    def __call__(self, src_cube):
+        """
+        Regrid this :class:`~iris.cube.Cube` on to the target grid of
+        this :class:`UnstructuredProjectedRegridder`.
+
+        The given cube must be defined with the same grid as the source
+        grid used to create this :class:`UnstructuredProjectedRegridder`.
+
+        Args:
+
+        * src_cube:
+            A :class:`~iris.cube.Cube` to be regridded.
+
+        Returns:
+            A cube defined with the horizontal dimensions of the target
+            and the other dimensions from this cube. The data values of
+            this cube will be converted to values on the new grid using
+            either nearest-neighbour or linear interpolation.
+
+        """
+        # Validity checks.
+        if not isinstance(src_cube, iris.cube.Cube):
+            raise TypeError("'src' must be a Cube")
+
+        src_x_coord, src_y_coord = get_xy_coords(src_cube)
+        tgt_x_coord, tgt_y_coord = self._tgt_grid
+        src_cs = src_x_coord.coord_system
+        tgt_cs = tgt_x_coord.coord_system
+
+        if src_x_coord.coord_system != src_y_coord.coord_system:
+            raise ValueError("'src' lateral geographic coordinates have "
+                             "differing coordinate sytems.")
+        if src_cs is None:
+            raise ValueError("'src' lateral geographic coordinates have "
+                             "no coordinate sytem.")
+
+        # Check the source grid units.
+        for coord in (src_x_coord, src_y_coord):
+            RectilinearRegridder._check_units(coord)
+
+        src_x_dim, = src_cube.coord_dims(src_x_coord)
+        src_y_dim, = src_cube.coord_dims(src_y_coord)
+
+        if src_x_dim != src_y_dim:
+            raise ValueError("'src' lateral geographic coordinates should map "
+                             "the same dimension.")
+        src_xy_dim = src_x_dim
+
+        # Compute the interpolated data values.
+        data = self._regrid(src_cube.data, src_xy_dim,
+                            src_x_coord, src_y_coord,
+                            tgt_x_coord, tgt_y_coord,
+                            self._projection, method=self._method)
+
+        # Wrap up the data as a Cube.
+        regrid_callback = functools.partial(self._regrid,
+                                            method=self._method,
+                                            projection=self._projection)
+
+        new_cube = self._create_cube(data, src_cube, src_xy_dim,
+                                     src_x_coord, src_y_coord,
+                                     tgt_x_coord, tgt_y_coord,
+                                     regrid_callback)
+
+        promote_aux_coord_to_dim_coord(new_cube, 'latitude')
+        promote_aux_coord_to_dim_coord(new_cube, 'longitude')
```

Comment:
Should be able to add the following to `_create_cube` after the calls to `copy_coords`:

```
promote_aux_coord_to_dim_coord(result, grid_x_coord)
promote_aux_coord_to_dim_coord(result, grid_y_coord)
```